src/Microsoft.Health.Fhir.Tests.Common/FhirException.cs

```diff
@@ -4,19 +4,18 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
-using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
-using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Core.Models;
 
-namespace Microsoft.Health.Fhir.Tests.E2E.Common
+namespace Microsoft.Health.Fhir.Tests.Common
```

Comment:
I think for E2E test, I think we should have STU3 and R4 projects with version specific tests implemented that and have them also include a shared project which hosts all of the common infrastructure and tests.

Most of the client will use something like the Firely .NET library and deserialize/deserialize resources directly and use HttpClient so I think it's beneficial to keep the test mimic that as opposed to use something like ResourceElement and introduce another layer that does conversions. 

What do you think?

Reply:
The point of the current test structure is to make use of the custom fixture `[HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.Json, FhirVersion.All)]`. Since the FhirVersion is essentially another vector of the type of test, I felt the approach in this PR make more sense.

Reply:
We can further review/get feedback, for now I like the `HttpIntegrationFixtureArgumentSets` approach also

Reply:
What I worry about is the readability of the code and maintenanbility of the test and the productivity of the dev who has to write new tests.

1. It moves away from the resource class so updating resource require new extension method to be written (if it doesn't already exist) and then update to FhirClient to be able to set these properties.
2. The logic is now located spread across different places (ModelExtensions, FhirClient, and the tests themselves).
3. Everything is ResourceElement now which is a bit hard to read. We would have to rely on the variable name to be able to tell what resource it's manipulating.
4. Accessing the properties of the resource becomes a bit hard since it has to rely on fhir path.

For E2E tests, we almost for certain will have to manipulate the resource directly to create objects with certain values. While it makes sense to consolidate into ResourceElement for the server code, I don't know if it makes that much sense for the E2E tests.